internal/pkg/dl/policies_integration_test.go

```diff
@@ -42,45 +41,81 @@ func createRandomPolicy(id string, revisionIdx int) model.Policy {
 	}
 }
 
-func storeRandomPolicy(ctx context.Context, bulker bulk.Bulk, index string) (model.Policy, error) {
-	var rec model.Policy
+func storeRandomPolicy(ctx context.Context, bulker bulk.Bulk, index string, maxRev int) error {
 	id := uuid.Must(uuid.NewV4()).String()
-	for i := 1; i < 4; i++ {
-		rec = createRandomPolicy(id, i)
-		_, err := CreatePolicy(ctx, bulker, rec, WithIndexName(index))
+	ops := make([]bulk.MultiOp, 0, maxRev)
+	for i := 1; i < maxRev; i++ {
+		rec := createRandomPolicy(id, i)
+		p, err := json.Marshal(&rec)
 		if err != nil {
-			return model.Policy{}, err
+			return err
 		}
+		ops = append(ops, bulk.MultiOp{
+			Index: index,
+			Body:  p,
+		})
 	}
-	return rec, nil
+	_, err := bulker.MIndex(ctx, ops, bulk.WithRefresh())
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
+//nolint:dupl // test duplication
 func TestQueryLatestPolicies(t *testing.T) {
 	ctx := testlog.SetLogger(t).WithContext(t.Context())
 
-	index, bulker := ftesting.SetupCleanIndex(ctx, t, FleetPolicies)
+	index, bulker := ftesting.SetupCleanIndex(ctx, t, FleetPolicies, bulk.WithFlushThresholdCount(1))
+
+	for i := 0; i < 4; i++ {
+		err := storeRandomPolicy(ctx, bulker, index, 4)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	time.Sleep(time.Second * 2) // FIXME ES does not refresh instantly?
+
+	policies, err := QueryLatestPolicies(ctx, bulker, WithIndexName(index))
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, policy := range policies {
+		if policy.RevisionIdx != 3 {
+			t.Errorf("Expected to find revision_idx 3 for policy %s, found %d", policy.PolicyID, policy.RevisionIdx)
+		}
+	}
+	if len(policies) != 4 {
+		t.Errorf("Expected 4 policies, got %d", len(policies))
+	}
+}
+
+//nolint:dupl // test duplication
+func TestQueryLatestPolicies300k(t *testing.T) {
+	ctx := testlog.SetLogger(t).WithContext(t.Context())
 
-	recs := map[string]model.Policy{}
-	for i := 0; i < 0; i++ {
-		rec, err := storeRandomPolicy(ctx, bulker, index)
+	index, bulker := ftesting.SetupCleanIndex(ctx, t, FleetPolicies, bulk.WithFlushThresholdCount(1))
+
+	for i := 0; i < 4; i++ {
+		err := storeRandomPolicy(ctx, bulker, index, 100000) // aggregation has a size limit of 10k, let's go over it
 		if err != nil {
 			t.Fatal(err)
 		}
-		recs[rec.PolicyID] = rec
 	}
+	time.Sleep(time.Second * 2) // FIXME ES does not refresh instantly?
 
 	policies, err := QueryLatestPolicies(ctx, bulker, WithIndexName(index))
```

Comment:
Maybe adding a `refresh` in the latest policies will result in it working with out the need for the sleep?

Reply:
Good suggestion, however it didn't work.

Reply:
Dang.